sqlx-postgres/src/types/point.rs

```diff
@@ -0,0 +1,140 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::types::Type;
+use crate::{PgArgumentBuffer, PgHasArrayType, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
+use sqlx_core::Error;
+use std::str::FromStr;
+
+const BYTE_WIDTH: usize = 8;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct PgPoint {
+    pub x: f64,
+    pub y: f64,
+}
+
+impl Type<Postgres> for PgPoint {
+    fn type_info() -> PgTypeInfo {
+        PgTypeInfo::with_name("point")
+    }
+}
+
+impl PgHasArrayType for PgPoint {
+    fn array_type_info() -> PgTypeInfo {
+        PgTypeInfo::with_name("_point")
+    }
+}
+
+impl<'r> Decode<'r, Postgres> for PgPoint {
+    fn decode(value: PgValueRef<'r>) -> Result<Self, Box<dyn std::error::Error + Send + Sync>> {
+        match value.format() {
+            PgValueFormat::Text => Ok(PgPoint::from_str(value.as_str()?)?),
+            PgValueFormat::Binary => Ok(pg_point_from_bytes(value.as_bytes()?)?),
+        }
+    }
+}
+
+impl<'q> Encode<'q, Postgres> for PgPoint {
+    fn produces(&self) -> Option<PgTypeInfo> {
+        Some(PgTypeInfo::with_name("point"))
+    }
+
+    fn encode_by_ref(&self, buf: &mut PgArgumentBuffer) -> Result<IsNull, BoxDynError> {
+        self.serialize(buf)?;
+        Ok(IsNull::No)
+    }
+}
+
+impl FromStr for PgPoint {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let sanitised = s.replace("(", "").replace(")", "").replace(" ", "");
+
+        let nums = sanitised.split(',').collect::<Vec<&str>>();
+
+        let x = nums
+            .first()
+            .map(|str| parse_float_from_str(str, "could not get x"))
+            .ok_or(Error::Decode("could not get x".into()))??;
+        let y = nums
+            .last()
+            .map(|str| parse_float_from_str(str, "could not get y"))
+            .ok_or(Error::Decode("could not get y".into()))??;
+
+        Ok(PgPoint { x, y })
+    }
+}
+
+fn pg_point_from_bytes(bytes: &[u8]) -> Result<PgPoint, Error> {
+    let x = get_f64_from_bytes(bytes, 0)?;
+    let y = get_f64_from_bytes(bytes, 8)?;
+    Ok(PgPoint { x, y })
+}
+
+impl PgPoint {
+    fn serialize(&self, buff: &mut PgArgumentBuffer) -> Result<(), Error> {
+        buff.extend_from_slice(&self.x.to_be_bytes());
+        buff.extend_from_slice(&self.y.to_be_bytes());
+        Ok(())
+    }
+
+    #[cfg(test)]
+    fn serialize_to_vec(&self) -> Vec<u8> {
```

Comment:
It's used for testing both serialise and deserialise

https://github.com/jayy-lmao/sqlx/blob/feat/support-geometry-postgres/sqlx-postgres/src/types/geometry/point.rs#L118

 https://github.com/jayy-lmao/sqlx/blob/feat/support-geometry-postgres/sqlx-postgres/src/types/geometry/point.rs#L136
 
It could be inlined in both tests and be different, but I found this easier.
Open to preference.